### PR TITLE
Update the fatal error message in starter.c

### LINKED ISSUE #1171: Apparent permissions error when loading SIF file in server
Hi everyone, I tried to squeeze as much info as possible in your template. If you need more, let me know, and thanks in advance. 

### Version of Apptainer

apptainer version 1.0.3+380-gea49318, installed on an HPC server, accessed with 'module load apptainer'. 

### Expected behavior

I want to pull a docker image, create s SIF file and use it to run my scripts. Pull works fine:

```
apptainer pull fd-test.sif docker://firedrakeproject/firedrake
```
Then, running a simple test script I have, I expect no output:
```
apptainer run --bind $PWD:/home/firedrake/work --home $PWD fd-test.sif /home/firedrake/firedrake/bin/python /home/firedrake/work/test.py
```

### Actual behavior

Instead, I get the following error:
```
ERROR  : Failed to retrieve root group information: Success
```

### What OS/distro are you running

```sh
CentOS 7 (Core)

```

### How did you install Apptainer

Preinstalled on system, loaded with modules interfaces.

### PR DESCRIPTION
**Pull request by JasonYangShadow**
## Description of the Pull Request (PR):

when errno = 0 (strerror(errno) will show SUCCESS), it'll add custom fatal message. otherwise use the errno. 

### This fixes or addresses the following GitHub issues:

 - Fixes #1171 

#### Before submitting a PR, make sure you have done the following:

- Read the [Guidelines for Contributing](https://github.com/apptainer/apptainer/blob/main/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- Added changes to the [CHANGELOG](https://github.com/apptainer/apptainer/blob/main/CHANGELOG.md) if necessary according to the [Contribution Guidelines](https://github.com/apptainer/apptainer/blob/main/CONTRIBUTING.md)
- Added tests to validate this PR, linted with `make check`  and tested this PR locally with a `make test`, and `make testall` if possible (see CONTRIBUTING.md).
- Based this PR against the appropriate branch according to the [Contribution Guidelines](https://github.com/apptainer/apptainer/blob/main/CONTRIBUTING.md)
- Added myself as a contributor to the [Contributors File](https://github.com/apptainer/apptainer/blob/main/CONTRIBUTORS.md)

### CODE REVIEW

**Comment by JasonYangShadow** [comment]
@DrDaveD I could not reproduce those issues 
https://github.com/apptainer/apptainer/issues/1171
I've deleted `root:x:0:` entry from /etc/group, I guess it is because I am using ubuntu:22.04 as the host os.

File: cmd/starter/c/starter.c [hunk 1]
```diff
@@ -1398,10 +1398,10 @@ __attribute__((constructor)) static void init(void) {
      * the host libc library used by this starter binary.
      */
     if (getpwuid(0) == NULL) {
-        fatalf("Failed to retrieve root user information: %s\n", strerror(errno));
+        fatalf("Failed to retrieve root user information: %s\n", errno == 0 ? "Not Found" : strerror(errno));
```
Comment (cclerget) [inline cmd/starter/c/starter.c:1401]: `Not found` => `root user not found`

File: cmd/starter/c/starter.c [hunk 2]
```diff
@@ -1398,10 +1398,10 @@ __attribute__((constructor)) static void init(void) {
...
     }
     if (getgrgid(0) == NULL) {
-        fatalf("Failed to retrieve root group information: %s\n", strerror(errno));
+        fatalf("Failed to retrieve root group information: %s\n", errno == 0 ? "Not Found" : strerror(errno));
```
Comment (cclerget) [inline cmd/starter/c/starter.c:1404]: `Not found` => `root group not found`

File: cmd/starter/c/starter.c [hunk 3]
```diff
@@ -1463,7 +1463,7 @@ __attribute__((constructor)) static void init(void) {
 
             /* wait parent write user namespace mappings */
             if ( wait_event(master_socket[1]) < 0 ) {
-                fatalf("Error while waiting event for user namespace mappings: %s\n", strerror(errno));
+                fatalf("Error while waiting event for user namespace mappings: %s\n", errno == 0 ? "No event from master socket" : strerror(errno));
```
Comment (cclerget) [inline cmd/starter/c/starter.c:1466]: `No event from master socket` => `no event received`

**Review by cclerget — DISMISSED** [review]

File: cmd/starter/c/starter.c (hunk 1 above)
Comment (JasonYangShadow) [reply cmd/starter/c/starter.c]: thanks! reflected here
https://github.com/apptainer/apptainer/pull/1483/commits/5e2eb22b69705515bd8a507b1598fa018d7015d4

File: cmd/starter/c/starter.c (hunk 2 above)
Comment (JasonYangShadow) [reply cmd/starter/c/starter.c]: thanks! reflected here
https://github.com/apptainer/apptainer/pull/1483/commits/5e2eb22b69705515bd8a507b1598fa018d7015d4

File: cmd/starter/c/starter.c (hunk 3 above)
Comment (JasonYangShadow) [reply cmd/starter/c/starter.c]: thanks! reflected here
https://github.com/apptainer/apptainer/pull/1483/commits/5e2eb22b69705515bd8a507b1598fa018d7015d4

**Comment by DrDaveD** [comment]
> @DrDaveD I could not reproduce those issues #1171 I've deleted `root:x:0:` entry from /etc/group, I am using ubuntu:22.04 as the host os.

The way to reproduce it was to chmod 600 /etc/group.

**Review by DrDaveD — APPROVED** [review]

**Comment by DrDaveD** [comment]
Also please cherry-pick to release-1.2

**Pull request merged**